index.ts

```diff
@@ -843,7 +843,7 @@ function validateFieldNames(validator: ValidatorJSON) {
 }
 
 function isValidIdentifier(name: string) {
-  return /^[a-zA-Z][a-zA-Z0-9_]*$/.test(name);
+  return /(^_(id|creationTime)$)|^[a-zA-Z][a-zA-Z0-9_]*$/.test(name);
```

Comment:
isValidIdentifier is used for things like table names and index names, maybe split this out into a separate isValidFieldName?